classes/Cart.php

```diff
@@ -4008,6 +4008,12 @@ public function getRawSummaryDetails(int $id_lang, bool $refresh = false): array
             'invoice' => AddressFormat::getFormattedLayoutData($invoice),
         ];
 
+        /**
+         * @see https://github.com/PrestaShop/PrestaShop/issues/25579#issuecomment-914269727
+         * we first get the cartRules with autoAdd = true.
+         * That will apply all applicable cartRules to the cart and then we calculate the totals and the discounts
+         */
+        $cartRules = $this->getCartRules();
```

Comment:
Maybe
```suggestion
        CartRule::autoAddToCart(...);
```
would make it more obvious that we do this, at this line, in order to apply to cart the cart rules?